node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala

```diff
@@ -126,11 +139,30 @@ case class P2PClientActor(
       case metaMsg: P2PClient.MetaMsg =>
         sender() ! handleMetaMsg(metaMsg)
       case ExpectResponseCommand(msg) =>
-        handleExpectResponse(msg)
+        Await.result(handleExpectResponse(msg), timeout)
       case Terminated(actor) if actor == peerConnection =>
         reconnect()
     }
 
+  private def ignoreNetworkMessages(
```

Comment:
Not all requests expect a response, it's only for `getheaders`, `getcfilters`, `getcfheaders` atm. But I get what you want to ask, in such a case both are processed. But only a matching response cancels the timer, so suppose I asked for headers and got a block, it will process the block and keep on waiting for headers.

Reply:
As for why and when we ignore messages, this only happens when we have started a disconnection from the peer. `initializeDisconnect` does not stop the actor in one actor message, it first sends a `Tcp.Close` to the `peerConnection` which when complete is intercepted by the actor and then the actor is stopped. The relevant part is that actor may process a message if its in the queue even while its disconnecting and that was causing some issues. We do not at all ignore any message in the usual operation.